CMakeLists.txt

```diff
@@ -3,6 +3,11 @@ include(src/cmake/projectVersionDetails.cmake)
 project(log4cxx VERSION ${log4cxx_VER} LANGUAGES CXX)
 include(CTest)
 
+# If you are including log4cxx from a higher-level CMake file(perhaps as a submodule?)
+# CMAKE_SOURCE_DIR will refer to the first CMakeLists.txt.  Since we need some files
+# in relation to the log4cxx source root, set a variable that contains the log4cxx root.
+set(LOG4CXX_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
The current implementation doesn't make that variable available in the GUI of Visual Studio:

![Bild_2021-04-17_111115](https://user-images.githubusercontent.com/6223655/115107849-a0070b80-9f6d-11eb-87f3-3deb8fbfcd82.png)

Changing that to the following line instead does and that approach is used for some other variables like e.g. `LOG4CXX_TEST_PROGRAM_PATH` as well.

    set(LOG4CXX_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}" CACHE PATH "Source root of log4cxx")
    set( LOG4CXX_TEST_PROGRAM_PATH "" CACHE PATH "Extra path for test programs" )

![Bild_2021-04-17_111654](https://user-images.githubusercontent.com/6223655/115108014-6a165700-9f6e-11eb-9378-a3f66ce6a7d5.png)

As this option is pretty important, shouldn't it be made as prominent as possible for people wanting to customize it?

Reply:
I do not believe any other value of LOG4CXX_SOURCE_DIR is going to work. 

Hence I suggest not making it available in the CMake GUI

Reply:
My understanding from Steven's email was that if you have a situation like the following:

```
/MyProject
/MyProject/CMakeLists.txt
/MyProject/log4cxx
/MyProject/log4cxx/CMakeLists.txt
```

where log4cxx is some kind of submodule to MyProject that you add with some CMake directive such as `add_directory`, then CMAKE_SOURCE_DIR will be set to /MyProject, which is an issue for how log4cxx finds certain files.  For example,  the ABI dump file(relative to the log4cxx root) is src/main/abi-symbols/abi.dump.  In this case, I think it would try to look in MyProject/src/main/abi-symbols/abi.dump, instead of the intended location of MyProject/log4cxx/src/main/abi-symbols/abi.dump.

Because of that, I don't believe that it needs to be configurable.  In the normal case of building log4cxx by itself, CMAKE_SOURCE_DIR will equal CMAKE_CURRENT_SOURCE_DIR at the top-level CMakeLists file.

I'm not positive about that though, so Steven's insight would be useful.

Reply:
I agree with Robert - it should not be configurable.

Reply:
I've simply didn't read correctly, `CMAKE_CURRENT_SOURCE_DIR` vs. `CMAKE_SOURCE_DIR`, I've read the latter instead of the former. What you say makes sense of course, no need to change things.